controllers/cloud.redhat.com/providers/resource_cache.go

```diff
@@ -121,22 +136,34 @@ func registerGVK(obj client.Object) {
 }
 
 // NewResourceIdent is a helper function that returns a ResourceIdent object.
-func NewSingleResourceIdent(provider string, purpose string, object client.Object) ResourceIdentSingle {
+func NewSingleResourceIdent(provider string, purpose string, object client.Object, opts ...ResourceOptions) ResourceIdentSingle {
 	registerGVK(object)
+	writeNow := false
+	for _, opt := range opts {
```

Comment:
maybe i'm not familiar with this code construct, but it feels like this allows for bizarre behavior where multiple options are passed in and you just take the value of the last one?

Reply:
I could wrap around saying only 1 is allowed to be passed

Reply:
It's a valid concern

Reply:
This is actually how k8s does it, if you supply two ListOptions it just loops through them one by one. I knew I'd seen the pattern somewhere - so yes thre is the potential confusion, but this seems to be standard pattern.